modules/core/src/main/java/org/apache/ignite/internal/table/criteria/SqlSerializer.java

```diff
@@ -161,12 +163,17 @@ private void append(String delimiter, String prefix, String suffix, Criteria[] e
      * Builder.
      */
     public static class Builder  {
+        private static final Pattern CANONICAL_NAME = Pattern.compile("^(?:\\p{Alpha}[\\w_]*\\.)?\\p{Alpha}[\\w_]*$");
```

Comment:
Seems, additional underscore is not needed.
> \w	A word character: [a-zA-Z_0-9]

Also, in case of simple name, moving the dot into a second group allows regexp matcher to not discard the first group.
```suggestion
        private static final Pattern CANONICAL_NAME = Pattern.compile("^(?:\\p{Alpha}[\\w]*)(?:\\.\\p{Alpha}[\\w]*)?$");
```

Reply:
Yes, you are right